### PR TITLE
Collect basic kubelet metrics (#1227

### PR DESCRIPTION
**Pull request by kwapik**
### Description of change
This adds a default user defined OTEL metrics pipeline, which collects basic kubelet metrics:
* `k8s.pod.cpu.utilization`
* `k8s.pod.memory.available`
* `k8s.pod.memory.usage`
* `k8s.pod.memory.working_set`

It also adds the following labels (if applicable):
* `k8s.daemonset.name`
* `k8s.cronjob.name`
* `k8s.deployment.name`
* `k8s.job.name`
* `k8s.namespace.name`
* `k8s.node.name`
* `k8s.pod.name`
* `k8s.pod.uid`
* `k8s.replicaset.name`
* `k8s.statefulset.name`

Closes: GH-1150

##### Checklist

- [x] Tested in playground or other setup
- [x] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added

- - -
This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1227)

### CODE REVIEW

File: pkg/otelcollector/config/builtin_config.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ func AddPrometheusRemoteWriteExporter(config *OTELConfig, promClient promapi.Cli
 	// tags, so they are not properly read by OTEL. Need to use bare maps instead.
 	config.AddExporter(otelconsts.ExporterPrometheusRemoteWrite, map[string]any{
 		"endpoint": endpoint.String(),
+		"resource_to_telemetry_conversion": map[string]any{
```
Comment (DariaKunoichi) [inline pkg/otelcollector/config/builtin_config.go:54]: Authors are planning to deprecate this option so we should be careful. Just saying.

**Review by DariaKunoichi — APPROVED** [review]

File: pkg/otelcollector/config/builtin_config.go (hunk 1 above)
Comment (kwapik) [reply pkg/otelcollector/config/builtin_config.go:54]: Created issue to track this https://github.com/fluxninja/aperture/issues/1228

**Comment by kwapik** [comment]
image
image
image

File: cmd/aperture-agent/agent/otel-config.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ type AgentOTELConfig struct {
 	BatchPostrollup BatchPostrollupConfig `json:"batch_postrollup"`
 	// CustomMetrics configures custom metrics pipelines, which will send data to
 	// the controller prometheus.
+	// By default `kubeletstats` custom metrics is added, which can be overwritten.
 	CustomMetrics map[string]CustomMetricsConfig `json:"custom_metrics,omitempty"`
```
Comment (tanveergill) [inline cmd/aperture-agent/agent/otel-config.go:42]: The key of `CustomMetrics` map refers to a pipeline name not metric name. Should clarify this in documentation as well as variable names in the code

**Comment by tanveergill** [comment]
I see code to enrich the metrics with service labels but I don't see service label in the

**Comment by tanveergill** [comment]
@kwapik,
Please also enable [k8sattributesprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor) in order to enrich the metrics with deployment name, labels etc.

File: cmd/aperture-agent/agent/otel-config.go (hunk 2 above)
Comment (kwapik) [reply cmd/aperture-agent/agent/otel-config.go:42]: Done.

**Comment by kwapik** [comment]
image

**Comment by kwapik** [comment]
image
It works also for other controllers, like `daemonset`:
```
k8s_pod_cpu_utilization{agent_group="default", k8s_daemonset_name="aperture-agent", k8s_namespace_name="aperture-agent", k8s_node_name="kind-worker2", k8s_pod_labels_app_kubernetes_io_component="aperture-agent", k8s_pod_labels_app_kubernetes_io_instance="agent", k8s_pod_labels_app_kubernetes_io_managed_by="aperture-operator", k8s_pod_labels_app_kubernetes_io_name="aperture", k8s_pod_name="aperture-agent-956rg", k8s_pod_uid="c86a52b5-ba6e-4fe4-a486-46fec354a495"}
```

**Pull request merged**